js/main.js

```diff
@@ -14,51 +12,81 @@ function storageAvailable(type) {
   } catch (e) {
     return e instanceof DOMException && (
       e.code === 22
-      || e.code === 1014
-      || e.name === 'QuotaExceededError'
-      || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+        || e.code === 1014
+        || e.name === 'QuotaExceededError'
+        || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
       && (storage && storage.length !== 0);
   }
 }
 
-function updateLocalStorage() {
-  // Updates the Local Storage
-  if (storageAvailable('localStorage')) {
-    localStorage.setItem('books', JSON.stringify(books));
-  }
-}
+class Book {
+  // Book Class has a title and author
+  title;
 
-const localBooksData = localStorage.getItem('books');
+  author;
 
-// Load initially stored data
-if (localBooksData) {
-  books = JSON.parse(localBooksData);
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
 }
 
-// Books Related Values & Functions
-const booksList = document.getElementById('books-list');
+class Library {
+  // Library class consists of a list of books
+  books = [];
 
-function bookExists(book) {
-  // Check if a book exists
-  for (let i = 0; i < books.length; i += 1) {
-    if (books[i].title === book.title && books[i].author === book.author) {
-      return true;
+  bookExists(book) {
+    // Check if a book exists
+    for (let i = 0; i < this.books.length; i += 1) {
+      if (this.books[i].title === book.title && this.books[i].author === book.author) {
+        return true;
+      }
     }
+    return false;
   }
-  return false;
-}
 
-function removeBook(book) {
-  // Remove a book from the list of lists
-  for (let i = 0; i < books.length; i += 1) {
-    if (books[i].title === book.title && books[i].author === book.author) {
-      books.splice(i, 1);
-      updateLocalStorage();
+  addBook(book) {
+    // Add a new book to the list of books
+    if (!this.bookExists(book)) {
+      // eslint-disable-next-line no-use-before-define
+      displayNewElement(book);
+      this.books.push(book);
+      this.updateLocalStorage();
       return;
     }
+    alert('The Book and Author exist');
+  }
+
+  removeBook(book) {
+    // Remove a book from the list of lists
+    for (let i = 0; i < this.books.length; i += 1) {
+      if (this.books[i].title === book.title && this.books[i].author === book.author) {
+        this.books.splice(i, 1);
+        this.updateLocalStorage();
+        return;
+      }
+    }
+  }
+
+  updateLocalStorage() {
+    // Updates the Local Storage
+    if (storageAvailable('localStorage')) {
+      localStorage.setItem('books', JSON.stringify(this.books));
+    }
   }
 }
 
+// Initialize the Library
+const library = new Library();
+
+// Load initially stored data
+const localBooksData = localStorage.getItem('books');
+if (localBooksData) {
+  library.books = JSON.parse(localBooksData);
+}
+
+// Books Related HTML Values & Functions
+const booksList = document.getElementById('books-list');
 function displayNewElement(book) {
```

Comment:
- for **ES6** good practice kindly use `arrow function` to declare your functions : 
```JS
const  displayNewElement = (book) => {...}
```

Reply:
I have changed this .. but can you please guide me as to why are arrow functions better?

Reply:
@Hamzaoutdoors  I have read that they make the syntax smaller if you have just one line of code, but in cases like above where you have multiple lines, why should they still be preferred?